app/controllers/mechanics_controller.rb

```diff
@@ -0,0 +1,12 @@
+class MechanicsController < ApplicationController
+  def show
+    @mechanic = Mechanic.find(params[:id])
+  end
+
+  def add_ride
+    mechanic = Mechanic.find(params[:id])
+    ride = Ride.find(params[:ride_id])
+    mechanic.rides << ride
+    redirect_to mechanic_path(mechanic)
```

Comment:
The logic here looks good, again, we just want to put it in a restful route, so because you have it here in the MechanicsController, I would have put this in an `def update` action and then in the routes, just opened the route for the update, but not had any nested routing.